### ISSUE TITLE
[4.0.6] Error on consequent usage of mix.js() and mix.react()

### ISSUE THREAD

**Issue by pahan35** [JeffreyWay/laravel-mix#1862]
-   Laravel Mix Version: 4.0.6 (`npm list --depth=0`)
-   Node Version (`node -v`): 8.12.0
-   NPM Version (`npm -v`): 6.4.1
-   Yarn Version (`yarn -v`): 1.12.3
-   OS: Windows 10 x64 Pro

### Description:
I noticed that consequent calls for `mix.js()` and `mix.react()` cause the error.
```
C:\xampp\htdocs\laravel-workshop\node_modules\webpack-cli\bin\cli.js:453
				throw err;
				^

Error: [VueLoaderPlugin Error] No matching use for vue-loader is found.
Make sure the rule matching .vue files include vue-loader in its use.
    at VueLoaderPlugin.apply (C:\xampp\htdocs\laravel-workshop\node_modules\vue-loader\lib\plugin.js:58:13)
    at webpack (C:\xampp\htdocs\laravel-workshop\node_modules\webpack\lib\webpack.js:47:13)
    at processOptions (C:\xampp\htdocs\laravel-workshop\node_modules\webpack-cli\bin\cli.js:441:16)
    at yargs.parse (C:\xampp\htdocs\laravel-workshop\node_modules\webpack-cli\bin\cli.js:536:3)
    at Object.parse (C:\xampp\htdocs\laravel-workshop\node_modules\yargs\yargs.js:567:18)
    at C:\xampp\htdocs\laravel-workshop\node_modules\webpack-cli\bin\cli.js:219:8
    at Object.<anonymous> (C:\xampp\htdocs\laravel-workshop\node_modules\webpack-cli\bin\cli.js:538:3)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (C:\xampp\htdocs\laravel-workshop\node_modules\webpack\bin\webpack.js:155:2)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Function.Module.runMain (module.js:694:10)
    at startup (bootstrap_node.js:204:16)
    at bootstrap_node.js:625:3
```

Looks like it happens due to overriding real resource tester by dummy true in view loader and when it when it calls second time to build assets it return fail matched url. The code is here `node_modules/vue-loader/lib/plugin.js:126`
![image](https://user-images.githubusercontent.com/13823215/49936626-18acb780-fedd-11e8-9ab5-50d2e5ca0944.png)

When I change it to `return false`, all works as expected, but I don't know which bugs I can achive.

### Steps To Reproduce:
You should create some react component and attach them in webpack.mix.js chain.
No bug
```
mix.js('resources/assets/js/app.js', 'public/js')
   .sass('resources/assets/sass/app.scss', 'public/css');
```

Bug
```
mix.js('resources/assets/js/app.js', 'public/js')
   .react('resources/assets/js/react-app.js', 'public/js')
   .sass('resources/assets/sass/app.scss', 'public/css');
```

You can take [my workshop repo](https://github.com/pahan35/laravel-workshop) (branch **error-on-consequent-call-js-and-react**) to reproduce the bug quickly.

Is it possible to fix this bug quickly?

**Comment by shahriar-b** [comment]
having same issue

**Comment by arcanedev-maroc** [comment]
I'm having the same issue but with `mix.ts(...)` with `laravel-mix: 4.0.7`.

BTW @JeffreyWay, where is the *documentation* about using laravel-mix with Typescript.

**Comment by JeffreyWay** [comment]
Fixed in 4.0.8.

@arcanedev-maroc I guess it hasn't been added yet. If you're willing, please submit a PR with TypeScript docs.
